internal/controller/pipelinerollout/pipelinerollout_ppnd.go

```diff
@@ -215,13 +224,13 @@ func (r *PipelineRolloutReconciler) pipelineNeedsUpdatingForPPND(ctx context.Con
 	specsEqual := util.CompareStructNumTypeAgnostic(fromCopy.Object["spec"], toCopy.Object["spec"])
 	numaLogger.Debugf("specsEqual: %t, from=%v, to=%v\n",
 		specsEqual, fromCopy.Object["spec"], toCopy.Object["spec"])
-	// compare Labels and Annotations, excluding any that Numaplane itself applies
-	labelsEqual := util.CompareMapsWithExceptions(from.GetLabels(), to.GetLabels(), common.KeyNumaplanePrefix)
-	numaLogger.Debugf("labelsEqual (excluding Numaplane labels): %t, from Labels=%v, to Labels=%v", labelsEqual, from.GetLabels(), to.GetLabels())
-	annotationsEqual := util.CompareMapsWithExceptions(from.GetAnnotations(), to.GetAnnotations(), common.KeyNumaplanePrefix)
-	numaLogger.Debugf("annotationsEqual (excluding Numaplane annotations): %t, from Annotations=%v, to Annotations=%v", annotationsEqual, from.GetAnnotations(), to.GetAnnotations())
+	// We need to restrict to just looking specifically at the labels and annotations we care about for data loss; otherwise, some platform (such as Numaflow) may set an annotation
+	// that we don't want to accidentally concern ourselves with
+	metadataRisk := usde.ResourceMetadataHasDataLossRisk(ctx, from, to)
+	numaLogger.Debugf("metadataRisk: %t, from=%v, to=%v\n",
+		metadataRisk, from.Object["metadata"], to.Object["metadata"])
 
-	return !specsEqual || !labelsEqual || !annotationsEqual, nil
+	return !specsEqual || metadataRisk, nil
```

Comment:
We need to make sure to ignore the new "numaflow.numaproj.io/resume-strategy" annotation is ignored when we're in PPND. We don't really need to be looking at all annotations and labels for determining if "pipeline needs to update" while we're in PPND mode. 

(Actually, it's on my TODO to see if we can just replace this function with something which just uses the existing USDE logic. I think the only reason it doesn't is because the idea was if we're already say pausing the pipeline due to isbsvc update, then if pipeline needs to update in any way (not just a data loss way), then we could simply update the pipeline now instead of waiting. But this may not be needed and could be made more simple).